federation/pkg/federation-controller/sync/controller.go

```diff
@@ -356,10 +356,34 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 		return statusError
 	}
 
+	schedObjsAccessor := func(adapter federatedtypes.FederatedTypeAdapter, fedObj pkgruntime.Object,
```

Comment:
nit: The reconcile function is 80 lines long.  Can we move this func out to reduce that by about 20 lines?